api/server/router/network/filter.go

```diff
@@ -29,6 +29,29 @@ func filterNetworkByType(nws []types.NetworkResource, netType string) ([]types.N
 	return retNws, nil
 }
 
+func filterNetworkByUse(nws []types.NetworkResource, danglingOnly bool) ([]types.NetworkResource, error) {
+	retNws := []types.NetworkResource{}
+
+	switch danglingOnly {
+	case true:
+		for _, nw := range nws {
+			if len(nw.Containers) == 0 && !runconfig.IsPreDefinedNetwork(nw.Name) {
```

Comment:
Now that https://github.com/docker/docker/pull/31710 is accepted, I'll finish this soon :+1: